### ISSUE TITLE
Is `variable_types` really optional?

### ISSUE THREAD

**Issue by iaindillingham** [opensafely-actions/cohort-report#23]
@sebbacon observed the following in [this Slack message](https://ebmdatalab.slack.com/archives/C0270Q313H7/p1632916489114700).

From README.md:

> The `generate_report` action outputs one HTML document with a graph for each variable specified. Where graphs have a category of 5 or less, small number suppression is applied and the whole graph is redacted.

Later on, however, `variable_types` is described as optional. How should one specify a variable without also specifying its type? Is `variable_types` really optional?

**Comment by iaindillingham** [comment]
Based on briefly 👀 *cohortreport/report.py*, I think:

* If `variable_types` isn't supplied, then tables and graphs are created for all columns in the input file
* If `variable_types` is supplied, then its values are used to apply column types, irrespective of the input file type
* If `variable_types` is supplied, then an `AssertionError` is raised when its keys don't match the columns in the input file

https://github.com/opensafely-actions/cohort-report/blob/715b7b8419fa964156e488078b1ffb97f57e9f1b/cohortreport/report.py#L64-L90

**Comment by sebbacon** [comment]
OK, so this sounds like reasonable behaviour as implemented, but there's a missing config step for just specifying a subset of columns without types. Perhaps a separate key `variables`, which is a list?

**Comment by iaindillingham** [comment]
Either that, or a type code/value that represents the existing type. An example of the former would be `"DEFAULT"`; of the latter would be `null`.

**Comment by sebbacon** [comment]
Yes. If we did that we should consider renaming the config key itself to something like `variables`
